lsstswBuild.sh

```diff
@@ -211,31 +212,51 @@ fi
 
 print_info "Rebuild is commencing....stand by; using $REF_LIST"
 
-RET=0
-# XXX REF_LIST and PRODUCT would be better handled as arrays
+ARGS=()
 if [ $NO_FETCH -eq 1 ]; then
-    "${LSSTSW}/bin/rebuild" -n "$REF_LIST" "$PRODUCT"
-    RET=$?
+    ARGS+=("-n")
 else
-    "${LSSTSW}/bin/rebuild" -u "$REF_LIST" "$PRODUCT"
-    RET=$?
+    ARGS+=("-u")
+fi
+if [[ ! -z "$REF_LIST" ]]; then
+    # XXX intentionally not quoted to allow word splitting
+    ARGS+=($REF_LIST)
+fi
+if [[ ! -z "$PRODUCT" ]]; then
+    # XXX intentionally not quoted to allow word splitting
+    ARGS+=($PRODUCT)
 fi
 
-# manifest.txt generated by lsst_build
-MANIFEST=${LSSTSW_BUILD_DIR}/manifest.txt
-
-# array of eups packages extracted from manifest; this excludes any repos that
-# may be checked out but not part of the current build graph
-PACKAGES=($(tail -n+3 "$MANIFEST" | awk '{ print $1 }'))
+"${LSSTSW}/bin/rebuild" "${ARGS[@]}"
+RET=$?
 
-# Set current build tag (also used as eups tag per installed package).
-eval "$(grep -E '^BUILD=' "$MANIFEST" | sed -e 's/BUILD/TAG/')"
+BUILD_PREPARED=false
+BUILD_FAILED=false
```

Comment:
It's a little confusing to have these with opposite senses.  Maybe BUILD_SUCCEEDED instead of BUILD_FAILED?